Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/PokePointer.cs

```diff
@@ -46,31 +83,23 @@ public override void OnPreSceneQuery()
 
             closestNormal = Rotation * Vector3.forward;
 
-            // Check proximity
-            BaseNearInteractionTouchable newClosestTouchable = null;
+            var layerMasks = PrioritizedLayerMasksOverride ?? PokeLayerMasks;
+
+            // Find closest touchable
+            ColliderNearInteractionTouchable newClosestTouchable = null;
+            foreach (var layerMask in layerMasks)
             {
-                closestDistance = float.PositiveInfinity;
-                foreach (var prox in BaseNearInteractionTouchable.Instances)
+                if (FindClosestTouchableForLayerMask(layerMask, out newClosestTouchable, out closestDistance, out closestNormal))
                 {
-                    if (prox.ColliderEnabled)
-                    {
-                        Vector3 normal;
-                        float dist = prox.DistanceToTouchable(Position, out normal);
-                        if (dist < prox.DistFront && dist < closestDistance)
-                        {   
-                            closestDistance = dist;
-                            newClosestTouchable = prox;
-                            closestNormal = normal;
-                        }
-                    }
+                    break;
                 }
             }
 
             if (newClosestTouchable != null)
             {
                 // Build ray (poke from in front to the back of the pointer position)
-                Vector3 start = Position - newClosestTouchable.DistBack * -closestNormal;
-                Vector3 end = Position + newClosestTouchable.DistFront * -closestNormal;
+                Vector3 start = Position - newClosestTouchable.PokeThreshold * -closestNormal;
```

Comment:
I actually think since you are not changing the RayStep here, that UnityUI is likely to work. You probably just need to add a UnityUITouchable component and set it up.

Reply:
I've tried that, and it doesn't work. As explained above, the problem is that we don't get here in the first place because the overlap test doesn't catch UI objects.